e2e/Uno/HelloUnoWorld.Wasm/HelloUnoWorld.Wasm.csproj

```diff
@@ -45,11 +43,11 @@
     <!-- Note that for WebAssembly version 1.1.1 of the console logger required -->
     <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="1.1.1" />
     <PackageReference Include="Microsoft.Extensions.Logging.Filter" Version="1.1.1" />
-    <PackageReference Include="Uno.Microsoft.Xaml.Behaviors.Uwp.Managed" Version="2.0.1-uno.28" />
-    <PackageReference Include="Uno.UI" Version="2.3.0-dev.2" />
-    <PackageReference Include="Uno.UI.RemoteControl" Version="2.3.0-dev.2" Condition="'$(Configuration)'=='Debug'" />
-    <PackageReference Include="Uno.Wasm.Bootstrap" Version="1.2.0-dev.24" />
-    <PackageReference Include="Uno.Wasm.Bootstrap.DevServer" Version="1.2.0-dev.24" />
+    <PackageReference Include="Uno.Microsoft.Xaml.Behaviors.Uwp.Managed" Version="2.0.2-uno.74" />
+    <PackageReference Include="Uno.UI.WebAssembly" Version="3.0.5" />
```

Comment:
@ebariche no need for Uno.UI + Uno.UI.WebAssembly at the time ?

Reply:
The Uno.UI build props/targets are accessible transitively, so you still get the proper build props/targets. I would assume that you do not need both, however there are some analyzers and additional resources that I'm not sure about. Once you both have determined whether as a best practice you only need a single `Uno.UI.WebAssembly` reference or if you also need the explicit `Uno.UI` reference and this conversation is marked as resolved, we'll merge the PR.

Reply:
Thanks @dansiegel !

Reply:
@agneszitte-nventive, it shouldn't be required, the E2E app works correctly !

Our new project template also doesn't have Uno.UI :

https://github.com/unoplatform/uno/blob/master/src/SolutionTemplate/UnoSolutionTemplate/Wasm/UnoQuickStart.Wasm.csproj

Let me get a second opinion from Carl !

Reply:
The change `Uno.UI` -> `Uno.UI.WebAssembly` is the **RIGHT CHANGE** for a Wasm app project.

![](https://media.giphy.com/media/MXX2jE0MChtYl1AgBe/giphy.gif)

Reply:
Missed that but the documentation is updated also @ebariche & @dansiegel ;)
https://github.com/unoplatform/uno/blob/665e3ff7f3ea386ae4f4d73f70aaf8898ab91a8e/doc/articles/migrating-from-previous-releases.md

Reply:
Doc: https://platform.uno/docs/articles/migrating-from-previous-releases.html

Reply:
thanks for the confirmation @carldebilly